schema/crdb/dbinit.sql

```diff
@@ -1885,49 +1885,76 @@ CREATE INDEX IF NOT EXISTS lookup_console_by_silo_user ON omicron.public.console
 
 /*******************************************************************/
 
-CREATE TYPE IF NOT EXISTS omicron.public.update_artifact_kind AS ENUM (
-    -- Sled artifacts
-    'gimlet_sp',
-    'gimlet_rot',
-    'host',
-    'trampoline',
-    'control_plane',
+-- Describes a single uploaded TUF repo.
+--
+-- Identified by both a random uuid and its SHA256 hash. The hash could be the
+-- primary key, but it seems unnecessarily large and unwieldy.
```

Comment:
First and foremost , I think the choice to use a UUID PK is totally reasonable here. We do so for many other objects that still have unique constraints elsewhere.

That being said: I had thought that the authz macros were the spot where a non-UUID PK was the most painful, but it seems like you've been able to work around that.

Isn't the "real" PK basically the system_version? as in, for a single "system version", shouldn't we have a single TUF repo, or am I mis-understanding this relationship?

Reply:
Yeah, the real primary key is the system version. But I think I was a bit concerned about an arbitrary-length DB primary key.